packages/grid/_modules_/grid/models/api/gridLocaleTextApi.ts

```diff
@@ -1,7 +1,12 @@
+import { Localization as CoreLocalization } from '@material-ui/core/locale';
+
+// @ts-expect-error We have different structure in v5
+type MuiComponentsLocalization = CoreLocalization['props'] | CoreLocalization['components'];
+
 /**
  * Set the types of the texts in the grid.
  */
-export interface GridLocaleText {
+export interface GridLocaleText extends Partial<MuiComponentsLocalization> {
```

Comment:
This type is not correct. 

1. This should fail: https://codesandbox.io/s/material-demo-forked-d8djr?file=/demo.tsx, it doesn't. Fixed in #2117.
2. The tests of https://github.com/mui-org/material-ui/pull/27196 shouldn't, they do. Fixed in #2117